static/js/latest-news.js

```diff
@@ -0,0 +1,112 @@
+function formatDate(date) {
+  const parsedDate = new Date(date);
+
+  const day = parsedDate.getDate();
+  const monthIndex = parsedDate.getMonth();
+  const year = parsedDate.getFullYear();
+
+  return year + "年" + (monthIndex + 1) + "月" + day + "日";
+}
+
+function htmlForLatestArticles(articles) {
+  const articlesTree = document.createDocumentFragment();
+  for (let i = 0; i < articles.length; i++) {
+    const article = articles[i];
+    const link = `<a href="/blog/${article.slug}">${
+      article.title.rendered
+    }</a>`;
+    const header = `<h4>${link}</h4>`;
+    const date = `<p class="u-no-padding--top">
+        <em>
+          <time pubdate datetime="${article.date}">
+            ${formatDate(article.date)}
+          </time>
+        </em>
+      </p>`;
+    const div = document.createElement("div");
+    div.classList.add("col-3");
+    div.innerHTML = ` 
+        ${header}
+        ${date}
+      `;
+
+    articlesTree.appendChild(div);
+  }
+  return articlesTree;
+}
+
+function htmlForLatestPinnedArticle(article) {
+  const articlesTree = document.createDocumentFragment();
+
+  // "Featured"
+  const header = "<h3>注目のニュース</h3>";
+
+  const link = `<a href="/blog/${article.slug}">${article.title.rendered}</a>`;
+  const linkHeader = `<h4>${link}</h4>`;
+
+  const date = `<p class="u-no-padding--top">
+      <em>
+        <time pubdate datetime="${article.date}">
+            ${formatDate(article.date)}
+        </time>
+      </em>
+    </p>`;
+
+  const innerDiv = `
+      ${linkHeader}
+      ${date}
+    `;
+
+  const div = document.createElement("div");
+  div.innerHTML = `
+    ${header}
+    ${innerDiv}
+  `;
+
+  articlesTree.appendChild(div);
+
+  return articlesTree;
+}
+
+const createReqListenerForDomContainer = (
+  latestNewsContainer,
+  spotlightContainer
+) => e => {
+  const data = JSON.parse(e.target.responseText);
+  const latest = data.latest_articles[0];
```

Comment:
There are quite some possibilities for this to fail and throw error.
`responseText` may be empty or not JSON, so `JSON.parse` would throw. Even if it is JSON it may not contain `latest_articles` for some reason (error in the network or backend).

If we don't have any specific error handling here it would be good to at least catch the errors. Not sure if jp.ubuntu.com has sentry where errors can be sent?

Reply:
I have added some `console.error` for now. I would have to investigate this. Maybe a good thing to extract into another issue?